src/taipy/core/data/sql.py

```diff
@@ -126,8 +130,15 @@ def storage_type(cls) -> str:
     def _get_base_read_query(self) -> str:
         return self.properties.get(self.__READ_QUERY_KEY)
 
+    def _do_append(self, data, engine, connection) -> None:
+        queries = self.properties.get(self._APPEND_QUERY_BUILDER_KEY)(data)
```

Comment:
We don't have guaranty that _APPEND_QUERY_BUILDER_KEY is provided in properties.

Reply:
I should and will raise an error if the attribute is not provided.